addons/html_editor/static/src/editor/link/link.js

```diff
@@ -0,0 +1,657 @@
+        // by the options but we still have to ensure that it is not removed if
+        // it exists in a template (e.g. "Newsletter Block" snippet).
+        if (!data.classes.split(' ').includes('btn')) {
+            for (const linkClass of this.toleratedClasses) {
+                if (this.state.iniClassName && this.state.iniClassName.split(' ').includes(linkClass)) {
+                    data.classes += " btn " + linkClass;
+                }
+            }
+        }
+        // When multiple buttons follow each other, they may break on 2 lines
+        // or more on mobile, so they need a margin-bottom.
+        if (data.classes.split(" ").includes("btn")) {
+            const closestButtonSiblingEls = this._getDirectButtonSiblings(this.linkEl);
+            if (closestButtonSiblingEls.length) {
+                data.classes += " mb-2";
+                closestButtonSiblingEls.forEach(btnEl => btnEl.classList.add("mb-2"));
+            }
+        }
+        if (['btn-custom', 'btn-outline-custom', 'btn-fill-custom'].some(className =>
+            data.classes.includes(className)
+        )) {
+            this.$link.css('color', data.classes.includes(data.customTextColor) ? '' : data.customTextColor);
+            this.$link.css('background-color', data.classes.includes(data.customFill) || isColorGradient(data.customFill) ? '' : data.customFill);
+            this.$link.css('background-image', isColorGradient(data.customFill) ? data.customFill : '');
+            this.$link.css('border-width', data.customBorderWidth);
+            this.$link.css('border-style', data.customBorderStyle);
+            this.$link.css('border-color', data.customBorder);
+        } else {
+            this.$link.css('color', '');
+            this.$link.css('background-color', '');
+            this.$link.css('background-image', '');
+            this.$link.css('border-width', '');
+            this.$link.css('border-style', '');
+            this.$link.css('border-color', '');
+        }
+        const attrs = Object.assign({}, this.state.oldAttributes, {
+            href: data.url,
+            target: data.isNewWindow ? '_blank' : '',
+        });
+        if (typeof data.classes === "string") {
+            data.classes = data.classes.replace(/o_default_snippet_text/, '');
+            attrs.class = `${data.classes}`;
+        }
+        if (data.rel) {
+            attrs.rel = `${data.rel}`;
+        }
+
+        this.$link.attr(attrs);
+        if (!this.$link.attr('target')) {
+            this.$link[0].removeAttribute('target');
+        }
+        this._updateLinkContent(this.$link, data);
+    }
+    /**
+     * Focuses the url input.
+     */
+    focusUrl() {
+        const urlInput = this.$el[0].querySelector('input[name="url"]');
+        urlInput.focus();
+        urlInput.select();
+    }
+
+    //--------------------------------------------------------------------------
```

Comment:
in owl component, we no longer need to make the distinction between private and non private methods. they are all private by default